apps/user/src/user/entities/user.entity.ts

```diff
@@ -1 +1,28 @@
-export class User {}
+import { CareerCategory } from '@apps/public-data/src/public-data/entities/career-category.entity';
+import { PositionCategory } from '@apps/public-data/src/public-data/entities/position-category.entity';
+import { ProjectCategory } from '@apps/public-data/src/public-data/entities/project-category.entity';
+import { Column, Entity, ManyToOne, PrimaryGeneratedColumn } from 'typeorm';
+
+@Entity('users')
+export class User {
+  @PrimaryGeneratedColumn()
+  public user_id: number;
+
+  @Column('varchar', { length: 255, unique: true, nullable: false })
+  public nickname: string;
+
+  @Column('varchar', { length: 255, unique: true, nullable: false })
+  public email: string;
+
+  @Column('simple-array')
+  public stack_category_id: number[];
+
+  @ManyToOne(() => PositionCategory, (position) => position.position_category_id)
+  public position_category_id: PositionCategory[];
```

Comment:
ManyToOne 관계일 때 position_category_id 필드의 타입이 배열인게 맞을까요?

Reply:
요렇게 되어야 하는게 아닐지 확인 부탁드립니다~
```suggestion
  @ManyToOne(() => PositionCategory, (position) => position.users)
  public position_category_id: PositionCategory[];
```

Reply:
제 PR에도 승완님이 리뷰해주신 내용인데, 테이터 타입 또한 배열이 아니어야 할 것 같습니다!

```typescript
public position_category_id: PositionCategory;
```

Reply:
[381cfdd](https://github.com/GrebbMe/grabbme-be/pull/37/commits/381cfdd71f1c4eab7d6ba1ddb49d61d85d8455dd)

말씀하신대로 관계정의에 대한 이해가 부족했던 것 같습니다!
users테이블은 position_category, project_category, career_category 에 대해 각 id를 `참조` 만 하는 형태여서,
```typescript
@ManyToOne(() => ProsionCategory, (position) => position.position_category_id)
```
의 코드가 아닌

```typescript
@ManyToOne(() => PositionCategory)
@JoinColumn({ name: 'position_category_id' })
public position_category_id: PositionCategory;
```
위와 같은 식으로 Entity의 관계정의를 했을 때, 올바르게 참조를 하는 것을 확인하였습니다!
![image](https://github.com/user-attachments/assets/942167ac-25e8-4fe6-a168-e71ba870f5b3)

결국 users 데이터는 FK 인 category 들의 정보를 사용하기만 할 뿐, category가 user의 정보를 가지는것이 아니기때문에
위와 같이 참조의 형태로만 선언하여 사용하는것이 올바르다고 판단하여
위의 커밋에 반영하였습니다

- 참고 : [TypeOrm Relation : ManyToOne, OneToMany](https://orkhan.gitbook.io/typeorm/docs/many-to-one-one-to-many-relations)